doc/Testing.md

```diff
@@ -0,0 +1,107 @@
+# Testing
+Improving test coverage is a great way to start contributing to Sia.  
+
+#### Table of Contents
+* [Run existing tests](#existing)
+  * [First update the code](#update)
+  * [Now run some tests](#run)
+* [Write more tests!](#write)
+  * [Naming conventions](#naming)
+  * [Table-driven tests](#table)
+  * [Benchmarks](#benchmarks)
+
+<a name="existing"/>
+## Run existing tests
+Go's comprehensive [test package][pkg/testing] makes testing straightforward, particularly when you use the bundled tools included in the [Sia makefile][makefile], including `make test`, `make cover`, `make bench`, and their variants.
+
+<a name="update"/>
+### First update the code
+```bash
+# Make sure you are in the right directory.
+$ cd $GOPATH/src/github.com/<your Github username>/Sia
+```
+
+If you want to test new code you've added, first make sure the rest of the code is up to date. New code should be on its own branch, e.g. `my_new_feature` (see ['Guide to Contributing to Sia.md'][guide]).
+```bash
+# Checkout the branch you made the changes on
+$ git checkout my_new_feature
+# Stash any tracked but uncommitted changes
+$ git stash
+# Then switch back to `master` and update it to match the original repo
+$ git checkout master
+$ git pull upstream master
+# Update `origin`, your fork of the repo
+$ git push origin master
+# Make the updated `master` the new base of `my_new_feature`,
+# which involves reapplying all the commits made to `my_new_feature`
+$ git checkout my_new_feature
+$ git rebase master --ignore-date
+# Restore the changes you stashed earlier
+$ git stash pop
+```
+When you call rebase, you may run into some merge conflicts.  Luke Champine's ['How to into git and GitHub'][luke] has more details (and many useful tricks).
+
+<a name="run"/>
+### Now run some tests
+The `make test` command runs all tests (functions starting with `Test` in `_test.go` files) for each package, setting off a panic for any test that runs longer than 5s.  For verbose output, run `make test-v` (which panics after 15s instead of 5s).  Finally, `make test-long` has verbose output, only panics when a test takes 5 minutes, and also cleans up your code using `gofmt` and `golint`.  You should run `make test-long` before each pull request.
+
+Run `make cover` to run all tests for each package and generate color-coded .html visualizations of test coverage by function for each source file.  Run `open cover/<module>.html` to inspect a module's test coverage in your default browser.
```

Comment:
`open` is OS X-specific. I don't think there's a generic POSIX command for this. Maybe just "open `cover/<module>.html` in your browser"